rpc/websocket_test.go

```diff
@@ -246,7 +249,7 @@ func wsPingTestHandler(t *testing.T, conn *websocket.Conn, shutdown, sendPing <-
 				t.Errorf("got pong with wrong data %q", data)
 			}
 			wantPong = ""
-			sendResponse = time.NewTimer(200 * time.Millisecond).C
+			sendResponse = timer.C
```

Comment:
This is not correct because the timer must be reset for every pong message.

Reply:
The idea in this test is: when a pong is received, the timer is started, and 200ms later a response is sent.

Reply:
Right, should be reset the timer every time when receive a pong

Reply:
It is still not quite correct now because the timer should not fire without a receive from `pongCh`. So you need to initialize the timer like 

```
timer := time.NewTimer(0)
defer timer.Stop()
<-timer.C
```

Reply:
I see your opinion, should new it once only pong is receive first time and defer stop when timer is not nil